url64700e9ab7_dwicke_course_scheduler_tgz-pJ8-org_json_TestJ8/src/coursescheduleapp/view/OrderedSchedView.java

```diff
@@ -0,0 +1,227 @@
+package coursescheduleapp.view;
+
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+import javax.swing.text.BadLocationException;
+import javax.swing.text.SimpleAttributeSet;
+import javax.swing.text.Style;
+import javax.swing.text.StyleConstants;
+import javax.swing.text.StyleContext;
+import javax.swing.text.StyledDocument;
+import javax.swing.text.html.HTML;
+
+import coursescheduleapp.model.Course;
+import coursescheduleapp.model.CourseData;
+import coursescheduleapp.model.Day;
+import coursescheduleapp.model.Schedule;
+import coursescheduleapp.model.TimeCollection;
+import coursescheduleapp.model.TimeDayCollection;
+import coursescheduleapp.model.TimePeriod;
+
+public class OrderedSchedView implements SchedViewI {
+
+	@Override
+	public void getResults(Course curr, StyledDocument doc) {
+		// TODO Auto-generated method stub
+
+		// Load the default style and add it as the "regular" text
+		Style def = StyleContext.getDefaultStyleContext().getStyle( StyleContext.DEFAULT_STYLE );
+
+		Style regular = doc.addStyle( "regular", def );
+
+		// Create a bold style
+		Style bold = doc.addStyle( "bold", regular );
+		StyleConstants.setBold( bold, true );
+		
+		
+
+		Map<String,Object> data = curr.getCourseInfo();
+
+		Set<String> keys = data.keySet();
+		Iterator<String> kIter = keys.iterator();
+
+		while(kIter.hasNext())
+		{
+			String key = kIter.next();
+
+			try {
+				
+
+				if (key == CourseData.INSTRUCTOR.toString())
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.